### PR TITLE
kubeadm: fix TestValidateConfig failure on the host with multiple CRI endpoints

### PR DESCRIPTION
**Pull request by SataQiu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
- Fix `TestValidateConfig` failure on the host with multiple CRI endpoints by skipping CRI detection for 'kubeadm config validate' command
- Cleanup UnknownCRISocket from static pods test

#### Which issue(s) this PR fixes:

Ref https://github.com/kubernetes/kubeadm/issues/2863

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/lgtm

**Comment by neolit123** [comment]
one way to check if we trigger auto-detection is to add e.g. a `panic()` in the auto-detection function and see if kubeadm trips if we run all unit tests. on a host that has a known socket the kubeadm tests will just pass, with autodetection working silently.

File: cmd/kubeadm/app/util/config/common.go [hunk 1]
```diff
@@ -332,21 +332,21 @@ func ValidateConfig(config []byte, allowExperimental bool) error {
 
 	// Validate InitConfiguration and ClusterConfiguration if there are any in the config
 	if kubeadmutil.GroupVersionKindsHasInitConfiguration(gvks...) || kubeadmutil.GroupVersionKindsHasClusterConfiguration(gvks...) {
-		if _, err := documentMapToInitConfiguration(gvkmap, true, allowExperimental, true, false); err != nil {
+		if _, err := documentMapToInitConfiguration(gvkmap, true, allowExperimental, true, true); err != nil {
```
Comment (SataQiu) [inline cmd/kubeadm/app/util/config/common.go:335]: For configuration verification, I don't think we need to actually perform the CRI check.
If the CRI Socket is empty, that should be a valid configuration and kubeadm will automatically populate it at run time. If the CRI Socket is not empty, that should also be a valid configuration, and kubeadm will respect the user's input.
WDYT?

Comment (neolit123) [reply cmd/kubeadm/app/util/config/common.go:335]: SGTM

**Review by neolit123 — COMMENTED** [review]
/lgtm

**Review by chendave — COMMENTED** [review]
/lgtm

**Pull request merged**